daemon/src/lib.rs

```diff
@@ -296,34 +296,45 @@ where
             Supervisor::new(move || ping::Actor::new(endpoint_addr.clone(), PING_INTERVAL));
         tasks.add(supervisor.run_log_summary());
 
-        let endpoint = Endpoint::new(
-            Box::new(TokioTcpConfig::new),
-            identity.libp2p,
-            ENDPOINT_CONNECTION_TIMEOUT,
-            TAKER_LISTEN_PROTOCOLS.inbound_substream_handlers(
-                pong_address.clone(),
-                identify_listener_actor,
-                libp2p_offer_addr,
-            ),
-            endpoint::Subscribers::new(
```

Comment:
consider adding a new EndpointRestarted message that invalidates the counts on all the observed connections / listeners

IMHO all the subscribers should be interested in that message, as this break the guarantees given earlier by the endpoint.

Reply:
My proposal is https://github.com/itchysats/itchysats/pull/2839/commits/0d1dd1bdb5cc12e38bfcf51dfc4d9cd43a58adf8.